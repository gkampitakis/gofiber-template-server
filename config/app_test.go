package config

import (
	"os"
	"testing"
	"time"

	"github.com/gkampitakis/fiber-modules/gracefulshutdown"
	"github.com/stretchr/testify/assert"
)

func clearEnv() {
	os.Unsetenv("GO_ENV")
	os.Unsetenv("APP_PORT")
	os.Unsetenv("APP_SHUTDOWN_PERIOD")
	os.Unsetenv("HC_TIMEOUT_PERIOD")
	os.Unsetenv("HC_TIMEOUT_ENABLED")
}

func TestAppConfig(t *testing.T) {
	t.Run("should return default values", func(t *testing.T) {
		defer clearEnv()
		clearEnv()

		config := New()
		assert.Equal(t, true, config.IsDevelopment)
		assert.Equal(t, "localhost:8080", config.Addr)
	})

	t.Run("should return explicit values and host as 0.0.0.0", func(t *testing.T) {
		defer clearEnv()
		clearEnv()
		os.Setenv("GO_ENV", "production")
		os.Setenv("APP_PORT", "1000")
		os.Setenv("GRACE_PERIOD", "35")

		fns := []func() error{}

		config := New()
		assert.Equal(t, false, config.IsDevelopment)
		assert.Equal(t, "0.0.0.0:1000", config.Addr)
		assert.Equal(t, gracefulshutdown.Config{
			Period:      time.Duration(35),
			Enabled:     true,
			ShutdownFns: fns,
		}, config.GracefulshutdownConfig(fns...))
	})
}
