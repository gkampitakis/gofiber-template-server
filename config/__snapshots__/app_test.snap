
[TestAppConfig/should_return_default_values - 1]
&config.AppConfig{
    _:                      struct {}{},
    Addr:                   "localhost:8080",
    Service:                "gofiber-template",
    IsDevelopment:          true,
    GracefulshutdownConfig: func(...func() error) gracefulshutdown.Config {...},
}
gracefulshutdown.Config{
    Period:             15,
    Enabled:            false,
    ShutdownFns:        nil,
    ListenErrorHandler: func(error) {...},
    Signals:            nil,
}
---

[TestAppConfig/should_return_explicit_values_and_host_as_0.0.0.0 - 1]
&config.AppConfig{
    _:                      struct {}{},
    Addr:                   "0.0.0.0:1000",
    Service:                "gofiber-template",
    IsDevelopment:          false,
    GracefulshutdownConfig: func(...func() error) gracefulshutdown.Config {...},
}
gracefulshutdown.Config{
    Period:             35,
    Enabled:            true,
    ShutdownFns:        {},
    ListenErrorHandler: func(error) {...},
    Signals:            nil,
}
---
